zproject/wsgi.py

```diff
@@ -15,14 +15,25 @@
 """
 import os
 import sys
+import logging
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 sys.path.append(BASE_DIR)
 import scripts.lib.setup_path_on_import
 
 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "zproject.settings")
 import django
-django.setup()  # We need to call setup to load applications.
+
+# It's common for a broken settings.py file to result in Django not
+# starting, thus never writing to errors.log.  To fix this, we simply
+# catch a NameError, if raised, and log it appropriately.
+logging.basicConfig(filename='/var/log/zulip/errors.log', level=logging.DEBUG,
+                    format='%(asctime)s %(levelname)s %(name)s %(message)s')
+logger = logging.getLogger(__name__)
+try:
+    django.setup()  # We need to call setup to load applications.
+except NameError as e:
+    logger.error(e)
```

Comment:
Can we put the block that configures the logger here, so it only runs in the exception case?  

Also, we should reraise the exception to exit, rather than continuing execution...

Reply:
(Also, I think if we use `logger.exception`, the full traceback will go in the log, which is probably what we want?)